controllers/vspheredeploymentzone_controller.go

```diff
@@ -112,7 +112,10 @@ func (r vsphereDeploymentZoneReconciler) Reconcile(ctx context.Context, request
 	failureDomain := &infrav1.VSphereFailureDomain{}
 	failureDomainKey := client.ObjectKey{Name: vsphereDeploymentZone.Spec.FailureDomain}
 	if err := r.Client.Get(ctx, failureDomainKey, failureDomain); err != nil {
-		if apierrors.IsNotFound(err) {
+		// If the VSphereFailureDomain is not found and the VSphereDeploymentZone is not deleting
+		// do not reconcile. If the VSphereDeploymentZone is in deletion allow reconcile to continue
```

Comment:
I think this does not work as documented. If we don't hit the if we hit the return in l.122

Reply:
Also `!vsphereDeploymentZone.DeletionTimestamp.IsZero()` is the deletion case. Not sure if I'm just misreading the godoc comment or the it's inverted

Reply:
Is this what you want?

```go
	if err := r.Client.Get(ctx, failureDomainKey, failureDomain); err != nil {
		if !apierrors.IsNotFound(err) {
			return reconcile.Result{}, err
		}
		
		if apierrors.IsNotFound(err) && vsphereDeploymentZone.DeletionTimestamp.IsZero() {
			logr.V(4).Info("Failure Domain not found, won't reconcile", "key", failureDomainKey)
			return reconcile.Result{}, nil
		}
	}
```

Reply:
(If we want to do it like this we also have to ensure reconcileDelete can handle this case, not sure if this is already the case)